cmake/legate_helper_functions.cmake

```diff
@@ -0,0 +1,384 @@
+macro(legate_include_rapids)
+  if (NOT _LEGATE_HAS_RAPIDS)
+    if(NOT EXISTS ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+      file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-22.08/RAPIDS.cmake
+           ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+    endif()
+    include(${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+    include(rapids-cmake)
+    include(rapids-cpm)
+    include(rapids-cuda)
+    include(rapids-export)
+    include(rapids-find)
+    set(_LEGATE_HAS_RAPIDS ON)
+  endif()
+endmacro()
+
+function(legate_default_cpp_install target)
+  set(options)
+  set(one_value_args EXPORT)
+  set(multi_value_args)
+  cmake_parse_arguments(
+    OPT
+    "${options}"
+    "${one_value_args}"
+    "${multi_value_args}"
+    ${ARGN}
+  )
+
+  if (NOT OPT_EXPORT)
+    message(FATAL_ERROR "Need EXPORT name for legate_default_install")
+  endif()
+
+  legate_include_rapids()
+
+  rapids_cmake_install_lib_dir(lib_dir)
+
+  install(TARGETS ${target}
+          DESTINATION ${lib_dir}
+	  EXPORT ${OPT_EXPORT})
+
+  set(code_string "set(legate_core_DIR ${legate_core_DIR})")
+
+  rapids_export(
+    INSTALL ${target}
+    EXPORT_SET ${OPT_EXPORT}
+    GLOBAL_TARGETS ${target}
+    NAMESPACE legate::
+    FINAL_CODE_BLOCK code_string
+  )
+
+  # build export targets
+  rapids_export(
+    BUILD ${target}
+    EXPORT_SET ${OPT_EXPORT}
+    GLOBAL_TARGETS ${target}
+    NAMESPACE legate::
+    FINAL_CODE_BLOCK code_string
+  )
+endfunction()
+
+function(legate_add_cffi header target)
+execute_process(
```

Comment:
Maybe want to error out if `CMAKE_C_COMPILER` is not set. That way users that have only enabled CXX or  CUDA don't get weird messages about unable to execute command `-E -P`